AmplifyPlugins/Core/AWSPluginsCore/AWSPluginOptions.swift

```diff
@@ -9,15 +9,53 @@ import Foundation
 import Amplify
 
 /// Plugin specific options type
-public struct AWSPluginOptions {
+///
+/// - Warning: Although this has `public` access, it is intended for internal use and should not be used directly
+///   by host applications. The behavior of this may change without warning.
+///   // OR SPI
+public struct AWSDataStorePluginOptions {
+
+    public static let DataStoreMetadataSource = "DataStore"
```

Comment:
I wish there was some surrounding context on this, can we add an extension to a type coming from Amplify Core, and add this static string? Something like
```swift
extension LazyReference {
  public static let DataStoreModelDecoderSource = "DataStore"
}
```